skore/src/skore/sklearn/_comparison/metrics_accessor.py

```diff
@@ -157,14 +164,175 @@ def report_metrics(
+    ) -> pd.DataFrame:
+        """Combine a list of dataframes.
+
+        Parameters
+        ----------
+        results : pd.DataFrame
+            The dataframes to combine.
+            They are assumed to originate from a `CrossValidationReport.metrics`
+            computation. In particular, there are several assumptions made:
+
+            - every dataframe has the form:
+                - index: Index "Metric", or MultiIndex ["Metric", "Label / Average"]
+                - columns: MultiIndex with levels ["Estimator", "Splits"] (can be
+                  unnamed)
+            - all dataframes have the same metrics
+
+            The dataframes are not required to have the same number of columns (splits).
+
+        estimator_names : list of str of len (len(results))
+            The name to give the estimator for each dataframe.
+
+        indicator_favorability : bool
+            Whether to keep the Favorability column.
+
+        aggregate : Aggregate
+            How to aggregate the resulting dataframe.
+        """
+
+        def add_model_name_to_index(df: pd.DataFrame, model_name: str) -> pd.DataFrame:
+            """Move the model name from the column index to the table index."""
+            df = copy.copy(df)
+
+            # Put the model name as a column
+            df["Estimator"] = model_name
+
+            # Put the model name into the index
+            if "Label / Average" in df.index.names:
+                new_index = ["Metric", "Label / Average", "Estimator"]
+            else:
+                new_index = ["Metric", "Estimator"]
+            df = df.reset_index().set_index(new_index)
+
+            # Then drop the model from the columns
+            df.columns = df.columns.droplevel(0)
+
+            return df
+
+        def melt_df(df):
+            df_reset = df.reset_index()
+
+            if "Label / Average" in df_reset.columns:
+                id_vars = ["Metric", "Label / Average", "Estimator"]
+            else:
+                id_vars = ["Metric", "Estimator"]
+
+            melted = pd.melt(
+                df_reset,
+                id_vars=id_vars,
+                var_name="Split",
+                value_name="Value",
+            )
+
+            return melted
+
+        results = individual_results.copy()
+
+        # Pop the favorability column if it exists, to:
+        # - not use it in the aggregate operation
+        # - later to only report a single column and not by split columns
+        if indicator_favorability:
+            favorability = results[0]["Favorability"]
+            for result in results:
+                result.pop("Favorability")
+        else:
+            favorability = None
+
+        dfs_model_name_in_index = [
+            add_model_name_to_index(df, estimator_name)
+            for df, estimator_name in zip(results, estimator_names)
+        ]
+
+        dfs_melted = [melt_df(df) for df in dfs_model_name_in_index]
+
+        df = pd.concat(dfs_melted, axis=0)
```

Comment:
I would not mind to chain the function to avoid having intermediate variable names.
```suggestion
        df = pd.concat([
            melt_df(add_model_name_to_index(df, estimator_name))
            for df, estimator_name in zip(results, estimator_names)
        ], axis=0)
```

Reply:
I agree, but the current phrasing makes debugging quite a bit easier. This whole function is ugly to me.

Reply:
`melt(df_with_name(df, estimator_name))` would be a nicer naming :)

I'm not convinced about the debugging because you will fail inside one of the function.

Reply:
applied suggestions. we can also debug with `breakpoint()`. Also renamed function